### PR TITLE
Add user download

### PR DESCRIPTION
**Pull request by rajikaimal**
This is work in progress !. Solves #116 
ps: Squashing the commit when merging would clean up the commit history which shows up here. :package: :tada: 

## Description

[WIP] Added `download` column to download completed jobs.

## Related Issue

#116 

## Screenshots (In case of UI changes):
screenshot from 2017-04-08 22 29 14

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by rajikaimal** [comment]
:construction:  **Work in progress** Let's consider about merging this once the work is complete. :tada:

**Comment by rajikaimal** [comment]
Files are now downloadable :tada: @rehrumesh 

**Changes made in the PR:** 

Endpoint _/api/download/<int>_ - is now checking for query parm _tk_ for the token, instead of the header since the Download is an _href_ according to UI changes.

UI - Added _Download_ href for downloading files in completed tab.

File: components/core/REST.py [hunk 1]
```diff
@@ -34,13 +34,13 @@ def token_validator(token):
...
-@server.route('/ui/<string:path>')
+@server.route('/ui/<path:path>')
 def serve_ui(path):
-    return send_from_directory(os.path.dirname(os.path.realpath(__file__))+"/ui", path)
+    return send_file(os.path.abspath(__file__ + "/../../../ui/dist/" + path))
 
 @server.route('/ui/')
 def serve_ui1():
-    return send_file(os.path.dirname(os.path.realpath(__file__))+"/ui/index.html")
+    return send_file(os.path.abspath(__file__ + "/../../../ui/dist/index.html"))
```
Comment (malithsen) [inline components/core/REST.py]: Let's get rid of this

**Comment by malithsen** [comment]
awesome work @rajikaimal 
Looks like you've worked on top of a previous PR which I closed. Let's revert it before merging :)

**Comment by agentmilindu** [comment]
@malithsen @rehrumesh Can we merge this PR?

**Comment by vivonk** [comment]
This work is done in GSoC-2018 Bassa project. Should we close this PR? @agentmilindu @rajikaimal @rehrumesh @malithsen

**Comment by agentmilindu** [comment]
@vivonk Yes, sure

**Pull request closed without merging**